app/scripts/modules/core/src/cluster/cluster.service.ts

```diff
@@ -213,10 +213,17 @@ export class ClusterService {
 
   private addNameParts(serverGroup: IServerGroup): void {
     const nameParts = this.namingService.parseServerGroupName(serverGroup.name);
-    serverGroup.app = nameParts.application;
-    serverGroup.stack = nameParts.stack;
-    serverGroup.detail = nameParts.freeFormDetails;
-    serverGroup.cluster = nameParts.cluster;
+    if (serverGroup.moniker) {
+      serverGroup.app = serverGroup.moniker.app;
+      serverGroup.stack = serverGroup.moniker.stack;
+      serverGroup.detail = serverGroup.moniker.detail;
+      serverGroup.cluster = serverGroup.moniker.cluster;
```

Comment:
Since all the moniker parts are being assigned to the server group, you could (I think, not sure if Typescript would complain) simplify these four lines to:
```javascript
Object.assign(serverGroup, serverGroup.moniker)
```